### PR TITLE
Feat/GDV(192/193)-integrate-vue2mapboxgl-and-default-theme

### PR DESCRIPTION
**Pull request by CindyvdVries**

### CODE REVIEW

File: layouts/default.vue [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
+      <v-mapbox
+        id="map"
+        ref="map"
+        :access-token="mapboxAccessToken"
+        map-style="mapbox://styles/global-data-viewer/cjtss3jfb05w71fmra13u4qqm"
+      >
+        <v-mapbox-navigation-control
+          position="bottom-right"
+        ></v-mapbox-navigation-control>
+        <v-mapbox-layer-clickable
```
Comment (SiggyF) [inline layouts/default.vue]: consider clickable versus raster by something based on existing domains.  For example feature/raster from gis or [sources](https://docs.mapbox.com/mapbox-gl-js/style-spec/#sources) from mapbox. Consider using [mixins](https://vuejs.org/v2/guide/mixins.html) for functionality that exists in multiple layer types.

File: lib/mapbox/layer-factory.js [hunk 2]
```diff
@@ -10,15 +10,7 @@ export default function layerFactory(sourceType, layer) {
...
-  const source = {
-    type: 'geojson',
-    data: {
-      type: 'FeatureCollection',
-      features: [],
-    },
-  }
-
-  return merge({ source }, layer)
+  return layer
```
Comment (SiggyF) [inline lib/mapbox/layer-factory.js]: remove function?

File: lib/mapbox/layers/get-locations-layer.js [hunk 3]
```diff
@@ -1,92 +1,55 @@
...
+  },
+  source: {
+    type: 'vector',
+    url: 'mapbox://global-data-viewer.6w19mbaw',
   },
   update(mapbox) {
     const source = mapbox.getSource(this.id)
-    const data = get('source.data', this)
-
+    const data = source ? source.data : undefined
```
Comment (SiggyF) [inline lib/mapbox/layers/get-locations-layer.js]: `_.get(source, 'data')`?

File: lib/mapbox/layers/get-locations-layer.js [hunk 4]
```diff
@@ -1,92 +1,55 @@
...
     } else {
       !source && console.warn(`Source with id ${this.id} could not be found`) // eslint-disable-line
       !data && console.warn('data property on layer could not be found', this) // eslint-disable-line
     }
   },
-  add(mapbox) {
-    const styleId = getCurrentStyle(mapbox).id
-    const defaultPaint = get('paint.default', this)
-    const paint = get(`paint[${styleId}]`, this)
+  get(mapbox) {
```
Comment (SiggyF) [inline lib/mapbox/layers/get-locations-layer.js]: Keep mapbox style and functionality separated?

File: test/unit/layouts/default.spec.js [hunk 5]
```diff
@@ -50,80 +46,62 @@ describe('Default', () => {
...
-    })
-
-    wrapper.find('select').element.value = 'dark'
-    wrapper.find('select').trigger('change')
-
-    expect(preferences.modules.theme.mutations.setActive.mock.calls[0][1]).toBe(
-      'dark',
-    )
-  })
+  // test('mounts without issues', () => {
```
Comment (SiggyF) [inline test/unit/layouts/default.spec.js]: remove

File: components/v-mapbox-components/v-mapbox-raster-layer.vue [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+          if (source && source.tiles[0] === newOptions.source.tiles[0]) return
+
+          if (source) {
+            if (layer) {
+              this.map.removeLayer(this.id)
+            }
+            this.map.removeSource(this.id)
+          }
+          if (get('options.source.tiles.length', this) > 0) {
```
[the remark this answers is not in the document]
Comment (CindyvdVries) [reply components/v-mapbox-components/v-mapbox-raster-layer.vue]: Empty strings can be added in backend if an error occurs or something else, default is then empty

Added a comment for this

File: lib/mapbox/layer-factory.js (hunk 2 above)
Comment (CindyvdVries) [reply lib/mapbox/layer-factory.js]: fix in: Feat-GDV-196-split-vector-layers-and-configuration

File: layouts/default.vue (hunk 1 above)
Comment (CindyvdVries) [reply layouts/default.vue]: fix in: Feat-GDV-196-split-vector-layers-and-configuration

**Pull request merged**